packages/nimble-components/src/rich-text/models/tests/markdown-parser.spec.ts

```diff
@@ -232,6 +232,93 @@ describe('Markdown parser', () => {
                 'Bulleted list with bold and italics'
             ]);
         });
+
+        it('numbers with escape character should be parsed as paragraph', () => {
+            const r = String.raw;
+            const doc = RichTextMarkdownParser.parseMarkdownToDOM(
+                r`1\.\ item 1
+                
+                2\. item 2
+                
+                3\.\item 3`
+            );
+
+            expect(getTagsFromElement(doc)).toEqual(['P', 'P', 'P']);
+            expect(getLeafContentsFromElement(doc)).toEqual([
+                r`1.\ item 1`,
+                r`2. item 2`,
+                r`3.\item 3`
+            ]);
+        });
+
+        it('bullet list with escape character should be parsed as paragraph', () => {
+            const r = String.raw;
+            const doc = RichTextMarkdownParser.parseMarkdownToDOM(
+                r`-\ item 1
+                
+                -\ item 2
+                
+                -\item 3`
+            );
+
+            expect(getTagsFromElement(doc)).toEqual(['P', 'P', 'P']);
+            expect(getLeafContentsFromElement(doc)).toEqual([
+                r`-\ item 1`,
+                r`-\ item 2`,
+                r`-\item 3`
+            ]);
+        });
+    });
+
+    describe('various escape characters should be parsed properly', () => {
+        const focused: string[] = [];
+        const disabled: string[] = [];
+        const r = String.raw;
+        const testsWithEscapeCharacters = [
+            { name: r`\*`, tags: ['P'], textContent: ['*'] },
+            { name: r`\*\*bold\*\*`, tags: ['P'], textContent: ['**bold**'] },
+            { name: r`\*italics\*`, tags: ['P'], textContent: ['*italics*'] },
+            { name: r`\# test1`, tags: ['P'], textContent: ['# test1'] },
+            {
+                name: r`\> blockquote`,
+                tags: ['P'],
+                textContent: ['> blockquote']
+            },
+            { name: r`\`code\``, tags: ['P'], textContent: ['`code`'] },
+            {
+                name: r`\~\~strikethrough\~\~`,
+                tags: ['P'],
+                textContent: ['~~strikethrough~~']
+            },
+            {
+                name: r`\## heading 2`,
+                tags: ['P'],
+                textContent: ['## heading 2']
+            },
+            {
+                name: r`\[link\](url)`,
+                tags: ['P'],
+                textContent: ['[link](url)']
+            }
+        ];
```

Comment:
Do we also need to extend this test case to assert hyphens, sequence of stars, underscores, etc. 

![image](https://github.com/ni/nimble/assets/94439533/c81fa9b4-b986-4941-bb90-a20fb61f9507)

Reply:
Will add them